File: AP1/Jutge/Consolidacio II/Patron.cc
#include <iostream>
using namespace std;

void patro(int x)
{
    if (x == 1)
        cout << 1 << endl;
    else {
        patro(x - 1);
        cout << x;
        for (int i = 1; i < x; ++i) {
            cout << " " << x;
        }
        cout << endl;
        patro(x - 1);
    }
}

int main()
{
    int n;
    cin >> n;
    patro(n);
}

File: AP1/Jutge/Consolidacio II/Patron.py
from easyinput import read

def patro(x):
    if x == 1:
        print("1")
    else:
        a = str(x);
        patro(x - 1)
        print(a, sep = "")
        patro(x - 1)

def main():
    n = read(int)
    patro(n)

main()
